### PR TITLE
FEATURE: Add `CanRender` Fusion prototype

### PR DESCRIPTION
**Pull request by mficzel**
The prototype checks wether the given type is renderable by fusion. Which means
that the prototype exists and has an implementation class or is based on a prototype that
has one.

### CODE REVIEW

**Comment by mficzel** [comment]
@dfeyer this is from https://github.com/ttreeagency/Fusion.CanRender

File: Neos.Fusion/Classes/FusionObjects/CanRenderImplementation.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * CanRender Fusion-Object
+ *
+ * The CanRender Fusion object checks wether the given type can be rendered
+ */
+class CanRenderImplementation extends AbstractFusionObject
+{
+    /**
+     * TypoScript Type which shall be rendered
```
Comment (hlubek) [inline Neos.Fusion/Classes/FusionObjects/CanRenderImplementation.php]: s/TypoScript/Fusion/

File: Neos.Fusion/Classes/FusionObjects/CanRenderImplementation.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+     *
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->tsValue('type');
+    }
+
+    /**
+     * Evaluate this TypoScript object and return the result
```
Comment (hlubek) [inline Neos.Fusion/Classes/FusionObjects/CanRenderImplementation.php]: s/TypoScript/Fusion/

File: Neos.Neos/Documentation/References/NeosFusionReference.rst [hunk 3]
```diff
@@ -517,6 +517,19 @@ Example::
...
 
+Neos.Fusion:CanRender
+---------------------
+
+Check wether a fusion prototype can be rendered. For beeing renderable a prototype must exist and have an
+implementation class. The implementation class can be defined indirectly via base prototypes.
+
+:type: (string) The prototype name that is checked
+
+   Example::
```
Comment (hlubek) [inline Neos.Neos/Documentation/References/NeosFusionReference.rst]: I think we don't need the indentation here

File: Neos.Neos/Documentation/References/NeosFusionReference.rst [hunk 4]
```diff
@@ -517,6 +517,19 @@ Example::
 		}
 	}
 
+Neos.Fusion:CanRender
+---------------------
+
+Check wether a fusion prototype can be rendered. For beeing renderable a prototype must exist and have an
```
Comment (hlubek) [inline Neos.Neos/Documentation/References/NeosFusionReference.rst]: s/wether/whether/

**Comment by kitsunet** [comment]
I would love if we would check the prototype and not a string, that would also retain prototype overrides, eg:

```
canRender = Neos.Fusion:CanRender {
    type = My.Package:Prototype
}
```

Note the missing quotes. Now type is an actual fusion prototype, so the ${path}/type could be checked for canRender. Although probably having the possibility to hand in a string (for dynamic purposes) is handy as well? Maybe both could work, but I guess not.

**Comment by mficzel** [comment]
@hlubek thanks for the nitpicks, i force pushed an update.

**Comment by mficzel** [comment]
@kitsunet using a prototype instead of a string would invalidate the intended use here https://github.com/neos/neos-development-collection/pull/1974#issuecomment-375605230 

This could also be used in the ContentCase to check wether the prototype exists and render a really helpful message if not.

**Comment by gerhard-boden** [comment]
@kitsunet Are you okay with handling the node type as a string? If so I would merge

**Pull request merged**